File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numsMap = new HashMap<>();
        
        // 키와 값을 바꿔서 맵으로 저장
        for(int i = 0; i < nums.length; i++) {
            numsMap.put(nums[i], i);
        }
        
        // target에서 첫 번째 수를 뺀 결과를 키로 조회, 현재 인덱스가 아닌 경루 정답으로 처리
        for(int i = 0; i < nums.length; i++) {
            if(numsMap.containsKey(target - nums[i]) && i != numsMap.get(target - nums[i])) {
                return new int[]{i, numsMap.get(target - nums[i])};
            }
        }
        
        // 항상 정답이 있으므로 null이 반환되는 경우는 없음
        return null;
    }
}

File: 0001-two-sum/0001-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val numsMap: MutableMap<Int, Int> = mutableMapOf()
        
        // 하나의 for문으로 통합
        for((i, num) in nums.withIndex()) {
            // target에서 num을 뺀 값이 있으면 정답으로 반환
            if(numsMap.containsKey(target - num)) {
                return intArrayOf(numsMap[target - num] ?: 0, i)
            }
            // 정답이 아니라면 값과 인덱스를 맵에 저장
            numsMap[num] = i
        }
        
        // 항상 정답이 존재하므로 반환하는 경우는 없음
        return intArrayOf(0, 0)
    }
}